numpy/core/src/umath/ufunc_object.c

```diff
@@ -575,13 +631,59 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
     int nout = ufunc->nout;
     int nop = ufunc->nargs;
     PyObject *obj, *context;
-    PyObject *str_key_obj = NULL;
     const char *ufunc_name = ufunc_get_name_cstr(ufunc);
     int type_num;
 
     int any_flexible = 0, any_object = 0, any_flexible_userloops = 0;
-    int has_sig = 0;
 
+    PyArray_Descr *dtype = NULL;
+    PyObject *sig = NULL;
+    typedef int (*converter)(PyObject *, void *);
```

Comment:
I often like to do my function typedefs as
```
typedef int converter(PyObject *, void *)
```

And then use `converter *convert` below, which reads as "a pointer to a converter function"

Reply:
I had mine from the CPython implementation, but that definitely looks nicer, so I changed it.